### PR TITLE
[MOOSE-130]: Misc Moose updates for WordPress 6.6 compatibility WIP

### PR DESCRIPTION
**Pull request by dpellenwood**
## What does this do/fix?
A variety of updates & improvements to Moose. Most of the changes are reflective of improvements WP Core has made with the release of v6.6. A few are just general bug fixes or tweaks to make Moose easier to use.

* `theme.json` objects should now be ordered alphabetically to male properties more easy to find.
* anywhere things in `theme.json` appear in a "size" array (font sizes, spacer sizes), the smallest size should always display first
* `theme.json` now allows you to style block style variations using the `variations` property - so some styling has been moved out of `.scss` files and into `theme.json` to reflect this.
* Query Pagination styles were somewhat confusing, it should now be properly nested so the hierarchy makes sense.
* Removes layout styling that we previously needed to override nested Group blocks
* theme.json version update and related changes. [Reference](https://make.wordpress.org/core/2024/06/19/theme-json-version-3/)
* Remove some old, commented out css from our reset that is no longer causing issues.
* root style selector updates per [Core's specificity updates](https://make.wordpress.org/core/2024/06/21/wordpress-6-6-css-specificity/).
* Remove the injectChanges parameter from BrowserSync config as css injection doesn't work.
* Update the `dist` npm script to use the `production` ENV value so that assets are minified.
* Add a `build` npm task to build non-minified assets on demand.
* Package updates for Composer & NPM (including plugins & WP Core).

### CODE REVIEW

**Comment by GeoffDusome** [comment]
This is a pretty huge PR but it was some _much needed_ updates to Moose as well as updates specific to changes made in WP 6.6. If you see some styling removed, it's mostly because it's been moved to `theme.json`. Alternatively, in some cases, the styling is no longer needed (the layout overrides). 

I'm going to bow out of reviewing this PR because I was pretty involved in it, but I'm interested what everyone else thinks of the updates.

**Review by Vinsanity — COMMENTED** [review]
🟢  Looks Great @dpellenwood 🌮  I added a comment that may be helpful enhancing our variable definitions.

**Review by LayaTaal — APPROVED** [review]
Works for me! I added a couple of minor comments which aren't blocking.

File: wp-content/themes/core/blocks/core/button/_variables.pcss [hunk 1]
```diff
@@ -2,15 +2,11 @@
...
- * NOTE: WordPress outputs their generated CSS properties on the body element.
- * To associate color properties w/ WP presets, and to allow style overrides
- * in the FSE Site Editor, CSS properties must be set on the body element.
- *
  * ------------------------------------------------------------------------- */
 
 /* @TODO: Add documentation on setting up theming to properly utilize */
 
-body {
+:root {
```
[the remark this answers is not in the document]
Comment (dpellenwood) [reply wp-content/themes/core/blocks/core/button/_variables.pcss:9]: That's an interesting idea. I'd like to explore theming in general a bit to see if we can make it more holistic. I added [a ticket to Moose](https://moderntribe.atlassian.net/browse/MOOSE-136) for us to look into it.

File: wp-content/themes/core/assets/pcss/typography/_mixins.pcss [hunk 2]
```diff
@@ -14,7 +14,7 @@
 	font-family: var(--wp--preset--font-family--roboto);
 	font-size: var(--wp--preset--font-size--80);
 	line-height: 1.1;
-	font-weight: 700;
+	font-weight: var(--wp--custom--font-weight--bold);
```
[the remark this answers is not in the document]
Comment (dpellenwood) [reply wp-content/themes/core/assets/pcss/typography/_mixins.pcss:17]: Nice! I [ticketed this](https://moderntribe.atlassian.net/browse/MOOSE-137) as a possible improvement.

**Pull request merged**